pallets/contracts/src/benchmarking.rs

```diff
@@ -563,4 +563,52 @@ benchmarks! {
     verify {
         assert_eq!(free_balance::<T>(&addr), ENDOWMENT + 1 as Balance);
     }
+
+    upgrade_api {
+        let alice_account_id: T::AccountId = UserBuilder::<T>::default().build("Alice").account();
+        let api_code_hash: ApiCodeHash<T> = ApiCodeHash { hash: CodeHash::<T>::default() };
+        let chain_version = ChainVersion::new(6, 0);
+        let api = Api::new(*b"POLY", 6);
+    }: _(RawOrigin::Root, alice_account_id.clone(), api.clone(), chain_version.clone(), api_code_hash.clone())
+    verify {
+        assert_eq!(
+            SupportedApiUpgrades::<T>::get(&alice_account_id, &api).get(&chain_version).unwrap(),
+            &api_code_hash
+        );
+    }
+
+    chain_extension_get_latest_api_upgrade {
+        let r in 0 .. CHAIN_EXTENSION_BATCHES;
+
+        let alice_account_id: T::AccountId = UserBuilder::<T>::default().build("Alice").account();
+        let api_code_hash: ApiCodeHash<T> = ApiCodeHash { hash: CodeHash::<T>::default() };
+        let output_len: u32 = api_code_hash.hash.as_ref().len() as u32;
+        let api = Api::new(*b"POLY", 6);
+
+        for i in 0..T::MaxApiUpgrades::get() {
+            Module::<T>::upgrade_api(
+                RawOrigin::Root.into(),
+                alice_account_id.clone(),
+                api.clone(),
+                ChainVersion::new(i, 0),
+                api_code_hash.clone(),
+            ).unwrap();
+        }
+
+        let encoded_input = (0..r * CHAIN_EXTENSION_BATCH_SIZE)
+            .map(|_| {
+                let mut input = alice_account_id.encode();
+                input.append(&mut api.encode());
+                input
+            })
+            .collect::<Vec<_>>();
+        let input_bytes =  encoded_input.iter().flat_map(|a| a.clone()).collect::<Vec<_>>();
```

Comment:
This input should be the `Api` encoded as bytes.  Could use `r` as the Api's version to provide different values.

Reply:
ah.  It needs to encode both an `AccountId` and `Api`.